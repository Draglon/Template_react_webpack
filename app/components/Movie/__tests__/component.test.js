import React from 'react';

import MovieComponent from '../component';

describe('<MovieComponent />', () => {
  const props = {
    movie: {
      id: 999,
      title: 'some title',
      overview: 'some overview',
      budget: 100000,
      genres: [1, 2, 3],
      revenue: 1000000,
      runtime: 140,
      language: 'English',
      backdrops: [{ file_path: 'path' }, { file_path: 'path' }, { file_path: 'path' }],
    },
    genres: [{ id: 1 }, { id: 2 }, { id: 3 }],
    cast: [1, 2, 3],
    crew: [1, 2, 3],
  };
  const component = shallow(<MovieComponent {...props} />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
