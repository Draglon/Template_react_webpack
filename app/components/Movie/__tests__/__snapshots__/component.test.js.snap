// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieComponent /> should match snapshot 1`] = `
<Fragment>
  <MovieCarouselComponent
    backdrops={
      Array [
        Object {
          "file_path": "path",
        },
        Object {
          "file_path": "path",
        },
        Object {
          "file_path": "path",
        },
      ]
    }
    title="some title"
  />
  <MovieTitleComponent
    movie={
      Object {
        "backdrops": Array [
          Object {
            "file_path": "path",
          },
          Object {
            "file_path": "path",
          },
          Object {
            "file_path": "path",
          },
        ],
        "budget": 100000,
        "genres": Array [
          1,
          2,
          3,
        ],
        "id": 999,
        "language": "English",
        "overview": "some overview",
        "revenue": 1000000,
        "runtime": 140,
        "title": "some title",
      }
    }
  />
  <MovieOverviewComponent
    overview="some overview"
    title="overview"
  />
  <MovieLabelComponent
    text="English"
    title="Original Language"
  />
  <MovieLabelComponent
    text="2h 20m"
    title="Runtime"
  />
  <MovieLabelComponent
    text="$100,000.00"
    title="Budget"
  />
  <MovieLabelComponent
    text="$1,000,000.00"
    title="Revenue"
  />
  <MovieGenresComponent
    genres={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
        Object {
          "id": 3,
        },
      ]
    }
    title="Genres"
  />
  <MovieDetailsComponent
    movieDetails={
      Array [
        1,
        2,
        3,
      ]
    }
    title="Casts"
  />
  <MovieDetailsComponent
    movieDetails={
      Array [
        1,
        2,
        3,
      ]
    }
    title="Crew"
  />
</Fragment>
`;
